### PR TITLE
docs: use proper error messages to fix examples a11y findings

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

This PR aims to fix accessibility review findings for the following components:

- [x] `DatePicker`
- [x] `DateTimePicker`
- [x] `TimePicker`

Fixes https://github.com/vaadin/web-components/issues/104
Fixes https://github.com/vaadin/web-components/issues/105
Fixes https://github.com/vaadin/web-components/issues/109

## Note

The part about old "Custom validator" example in https://github.com/vaadin/web-components/issues/104 is no longer relevant:

> The same happens for the [Custom validator](https://vaadin.com/components/date-picker/html-examples/date-picker-validation-demos#custom-validator) example if the user chooses a date that is not a working day.

> The date picker with an invalid date - only the background colour of the control is changed

### CODE REVIEW

File: frontend/demo/component/timepicker/time-picker-min-max.ts [hunk 1]
```diff
@@ -24,6 +28,17 @@ export class Example extends LitElement {
         min="08:00"
         max="16:00"
         .step="${60 * 30}"
+        error-message="${this.errorMessage}"
+        @change="${(event: TimePickerChangeEvent) => {
+          const { min, max, value } = event.target;
+          if (value < min) {
```
Comment (vursen) [inline frontend/demo/component/timepicker/time-picker-min-max.ts:34]: Are you comparing strings here? I'm afraid this won't work.

Comment (web-padawan) [reply frontend/demo/component/timepicker/time-picker-min-max.ts:34]: It actually works at least in such simple cases. I could add a comment about it though.

Screenshot 2023-02-16 at 13 22 45

Comment (vursen) [reply frontend/demo/component/timepicker/time-picker-min-max.ts:34]: Hmm, right.

File: frontend/demo/component/datepicker/date-picker-min-max.ts [hunk 2]
```diff
@@ -36,6 +41,19 @@ export class Example extends LitElement {
         .max="${this.upperLimit}"
         label="Appointment date"
         helper-text="Must be within 60 days from today"
+        .errorMessage="${this.errorMessage}"
+        @change="${({ target }: DatePickerChangeEvent) => {
+          const [value, min, max] = [target.value, this.today, this.upperLimit].map((date) =>
```
Comment (vursen) [inline frontend/demo/component/datepicker/date-picker-min-max.ts:46]:
```suggestion
          const [value, min, max] = [target.value, this.minDate, this.maxDate].map((date) =>
```

Comment (vursen) [reply frontend/demo/component/datepicker/date-picker-min-max.ts:46]: How about we align the way we define `minDate` and `maxDate` with the DateTimePicker example below?

File: src/main/java/com/vaadin/demo/component/datetimepicker/DateTimePickerMinMax.java [hunk 3]
```diff
@@ -20,6 +20,18 @@ public DateTimePickerMinMax() {
         dateTimePicker.setMin(LocalDateTime.now());
         dateTimePicker.setMax(LocalDateTime.now().plusDays(60));
         dateTimePicker.setValue(LocalDateTime.now().plusDays(7));
+        dateTimePicker.addValueChangeListener(event -> {
+            LocalDateTime value = event.getValue();
+            if (value.compareTo(dateTimePicker.getMin()) < 0) {
```
Comment (vursen) [inline src/main/java/com/vaadin/demo/component/datetimepicker/DateTimePickerMinMax.java:25]: You should probably call `compareTo` on `dateTimePicker.getMin()` because `value` can be `null`.

File: frontend/demo/component/datepicker/date-picker-min-max.ts (hunk 2 above)
Comment (web-padawan) [reply frontend/demo/component/datepicker/date-picker-min-max.ts]: Done.

File: src/main/java/com/vaadin/demo/component/datetimepicker/DateTimePickerMinMax.java (hunk 3 above)
Comment (web-padawan) [reply src/main/java/com/vaadin/demo/component/datetimepicker/DateTimePickerMinMax.java]: Added `null` checks in listeners.

**Review by vursen — APPROVED** [review]

File: frontend/demo/component/datetimepicker/date-time-picker-custom-validation.ts [hunk 4]
```diff
@@ -19,17 +19,19 @@ export class Example extends LitElement {
   // tag::snippet[]
   private binder = new Binder(this, AppointmentModel);
 
+  private errorMessage = 'The selected day of week or time is not available';
+
```
Comment (tomivirkki) [inline frontend/demo/component/datetimepicker/date-time-picker-custom-validation.ts:23]: Why this change? The example demonstrated that you can have different error messages for different condition violations. Not both condition violations show the same error message.

Comment (web-padawan) [reply frontend/demo/component/datetimepicker/date-time-picker-custom-validation.ts:23]: @tomivirkki This is a TetraLogical finding from #105: 

> For the Custom Validation example, if the user chooses both an invalid date and an invalid time, an error text is shown that only mentions the date, meaning the error for the time field is only indicated using colour.

We already have both DatePicker and TimePicker using different error messages depending on min / max.
So I thought it should be "good enough" this way to fully address the original accessibility issues.

Comment (tomivirkki) [reply frontend/demo/component/datetimepicker/date-time-picker-custom-validation.ts:23]: Ah, I see. Makes sense :+1: For Hilla users, there still the `custom-field-basic` example, which has multiple constraints with different error messages.

**Review by tomivirkki — APPROVED** [review]

**Comment by web-padawan** [comment]
Changed the base branch to `latest` so that these fixes also make it to the Vaadin 23.3 documentation.
